pulp_2to3_migration/app/pre_migration.py

```diff
@@ -535,10 +535,13 @@ def pre_migrate_distributor(repo_id, distributors, distributor_migrators, repo=N
                       'pulp2_type_id': dist_data.distributor_type_id,
                       'pulp2_last_updated': last_updated,
                       'pulp2_config': dist_data.config,
-                      'pulp2_repository': repo,
                       'pulp2_repo_id': repo_id,
                       'is_migrated': False})
 
+        if repo:
```

Comment:
@goosemania I started to go the way we have discussed on the call and then I have realized that we have probably confused ourselves. I have not added anything.
Simple plan- the relations will be handled here, because in simple plan we always have 1:1 repo vs distributor, so we do not need to add anything else in the `simple_plugin_migration`
Complex plan - if you happened to specify a native distributor from the repo you are trying to migrate, it will be added here, and then later in `complex_repo_migration`

Reply:
I agree that simple one is going to work.
I have a new concern. I think we need to identify if it's a simple plan and only then add the native distributor.

Imagine a complex plan with swapped distributors:
- for repo A use distributor B
- for repo B use distributor A

I think with the current approach both repos will have both distributors which is wrong.
So I propose to add a native distributor only for simple plan:
```
if distributors and repo:
    repo.pulp2_dists.add(distributor)
```
Thoughts?

Reply:
did you mean :
```
if not distributors and repo:
    repo.pulp2_dists.add(distributor)
```
In case it is a simple plan distributors will be [], but in case it is a complex plan it will contain all the distributors specified in it. This is a fine approach for now, but once we get done simple-complex plan conversion work, it will no longer work, which is fine. We just need to leave a note to remove these 2 lines.

Another option where to add the relations in case of simple plan would be in `simple_plugin_migration`
```
$ git diff
diff --git a/pulp_2to3_migration/app/migration.py b/pulp_2to3_migration/app/migration.py
index 8380e14..680c604 100644
--- a/pulp_2to3_migration/app/migration.py
+++ b/pulp_2to3_migration/app/migration.py
@@ -189,6 +189,8 @@ def simple_plugin_migration(plugin):
     for pulp2_dist in pulp2distributors_qs:
         dist_migrator = distributor_migrators.get(pulp2_dist.pulp2_type_id)
         migrate_repo_distributor(dist_migrator, progress_dist, pulp2_dist)
+        pulp2_repo = repos_to_migrate.filter(pulp2_repo_id=pulp2_dist.pulp2_repo_id)[0]
+        pulp2_repo.pulp2_dists.add(pulp2_dist)
 
```
This piece of code ^ will entirely go away after simple-complex plan conversion

I think I prefer the first option.

Reply:
@jlsherrill 1)when you create a complex migration plan is it a normal case when you use a distributor from a different repo, e.g. distribute repoA with the ditributor from repoB?
2) is it supported to swap distributors/importers between the migrations? for example in first run you distribute repoA with the distributor from repoB and repoB with the distributor from repoA, and in the second run(in the new plan) you distribute repoA with the distributor from repoA and repoB with the distributor from repoB?

Reply:
The situation you describe would not happen, however, we could have:

Repo A
Repo B

and ask you to distribute the repo from Repo A with the distributor from Repo B.  Then on the next migration we have:

Repo A
Repo B
Repo C

and now we ask you to distribute Repo C with the distributor from Repo B.

Reply:
> The situation you describe would not happen, however, we could have:
> 
> Repo A
> Repo B
> 
> and ask you to distribute the repo from Repo A with the distributor from Repo B. Then on the next migration we have:
> 
> Repo A
> Repo B
> Repo C
> 
> and now we ask you to distribute Repo C with the distributor from Repo B.

And what do you ask to distribute repo A with? some other distributor or repo A is no longer in the plan?

Reply:
It could still be in the plan, we could ask for it to be distributed via Repo D's distributor.

Reply:
@ipanova , let's merge and file this as a separate issue. WDYT?